File: backend/src/main/java/com/soulh/model/Report.java
package com.soulh.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Document(collection = "reports")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Report {
    @Id
    private String id;
    private String reporterId;
    private String reportedUserId;
    private String reason; // HARASSMENT, SPAM, INAPPROPRIATE, OTHER
    private String description;
    private String contextMessageId; // Optional: specific message that triggered the report
    
    @Builder.Default
    private boolean resolved = false;
    
    private LocalDateTime createdAt;
    
    public void onCreate() {
        this.createdAt = LocalDateTime.now();
    }
}

File: node-backend/models/Report.js
const mongoose = require('mongoose');

const ReportSchema = new mongoose.Schema({
  reporterId: {
    type: String,
    required: true,
    index: true
  },
  reportedUserId: {
    type: String,
    required: true,
    index: true
  },
  reason: {
    type: String,
    required: true
  },
  description: {
    type: String
  },
  contextMessageId: {
    type: String
  },
  resolved: {
    type: Boolean,
    default: false
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
});

// toJSON transform helper
ReportSchema.set('toJSON', {
  transform: (doc, ret) => {
    ret.id = ret._id.toString();
    delete ret._id;
    delete ret.__v;
    return ret;
  }
});

module.exports = mongoose.model('Report', ReportSchema);
